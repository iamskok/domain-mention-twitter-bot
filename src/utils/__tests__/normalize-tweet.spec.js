import normalizeTweet from '../normalize-tweet';
import rawTweets from '../../tests/mocks/rawTweets';

describe('normalizeTweet', () => {
  it('returns normalized tweet', () => {
    expect(normalizeTweet(rawTweets[0])).toMatchSnapshot();
  });

  it('returns normalized tweet reply', () => {
    expect(normalizeTweet(rawTweets[1])).toMatchSnapshot();
  });

  it('returns normalized tweet quote', () => {
    expect(normalizeTweet(rawTweets[2])).toMatchSnapshot();
  });
});
