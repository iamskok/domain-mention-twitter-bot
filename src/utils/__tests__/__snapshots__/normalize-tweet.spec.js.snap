// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeTweet returns normalized tweet 1`] = `
Object {
  "created_at": "Sat Nov 14 18:14:07 +0000 2020",
  "entities": Object {
    "hashtags": Array [],
    "symbols": Array [],
    "urls": Array [
      Object {
        "display_url": "skok.club",
        "expanded_url": "http://skok.club",
        "indices": Array [
          0,
          23,
        ],
        "url": "https://t.co/4qg8i7N2NZ",
      },
    ],
    "user_mentions": Array [],
  },
  "extended_tweet": Object {
    "full_text": "https://t.co/4qg8i7N2NZ tweet",
  },
  "id_str": "1327676200005869570",
  "in_reply_to_status_id_str": null,
  "quoted_status_id_str": null,
  "quotes": null,
  "replies": null,
  "user": Object {
    "name": "Pavel K",
    "profile_image_url_https": "https://abs.twimg.com/sticky/default_profile_images/default_profile_400x400.png",
    "screen_name": "bbk136",
  },
}
`;

exports[`normalizeTweet returns normalized tweet quote 1`] = `
Object {
  "created_at": "Sat Nov 14 18:12:48 +0000 2020",
  "entities": Object {
    "hashtags": Array [],
    "symbols": Array [],
    "urls": Array [
      Object {
        "display_url": "skok.club",
        "expanded_url": "http://skok.club",
        "indices": Array [
          0,
          23,
        ],
        "url": "https://t.co/4qg8i7N2NZ",
      },
    ],
    "user_mentions": Array [],
  },
  "extended_tweet": Object {
    "full_text": "https://t.co/4qg8i7N2NZ quote",
  },
  "id_str": "1327675869255643136",
  "in_reply_to_status_id_str": null,
  "quoted_status_id_str": "1305258930394861570",
  "quotes": null,
  "replies": null,
  "user": Object {
    "name": "Pavel K",
    "profile_image_url_https": "https://abs.twimg.com/sticky/default_profile_images/default_profile_400x400.png",
    "screen_name": "bbk136",
  },
}
`;

exports[`normalizeTweet returns normalized tweet reply 1`] = `
Object {
  "created_at": "Sat Nov 14 18:11:31 +0000 2020",
  "entities": Object {
    "hashtags": Array [],
    "symbols": Array [],
    "urls": Array [
      Object {
        "display_url": "skok.club",
        "expanded_url": "http://skok.club",
        "indices": Array [
          14,
          37,
        ],
        "url": "https://t.co/4qg8i7N2NZ",
      },
    ],
    "user_mentions": Array [
      Object {
        "id": 837337609974112300,
        "id_str": "837337609974112256",
        "indices": Array [
          0,
          13,
        ],
        "name": "San Diego CPA",
        "screen_name": "CPA_SanDiego",
      },
    ],
  },
  "extended_tweet": Object {
    "full_text": "@CPA_SanDiego https://t.co/4qg8i7N2NZ test",
  },
  "id_str": "1327675544016723969",
  "in_reply_to_status_id_str": "1305258930394861570",
  "quoted_status_id_str": null,
  "quotes": null,
  "replies": null,
  "user": Object {
    "name": "Pavel K",
    "profile_image_url_https": "https://abs.twimg.com/sticky/default_profile_images/default_profile_400x400.png",
    "screen_name": "bbk136",
  },
}
`;
